internal/proto/codec/builtin.go

```diff
@@ -631,11 +631,11 @@ func DecodeListMultiFrameForData(frameIterator *proto.ForwardFrameIterator) []is
 	return result
 }
 
-func DecodeListMultiFrameWithListInteger(frameIterator *proto.ForwardFrameIterator) []int32 {
-	result := make([]int32, 0)
+func DecodeListMultiFrameWithListInteger(frameIterator *proto.ForwardFrameIterator) [][]int32 {
+	result := make([][]int32, 0)
```

Comment:
Great fix! It would better to use var initializer for default construction and also to use the existing `DecodeListMultiFrame` functionality iterate the frames.
```suggestion
	var result [][]int32
	DecodeListMultiFrame(frameIterator, func(fi *proto.ForwardFrameIterator) {
		result = append(result, DecodeListInteger(fi))
	})
	return result
```

Reply:
Sorry, I believe I misplaced the suggestion. Can you refactor the method to eliminate unreachable code?